core/src/main/java/software/amazon/awssdk/retry/SdkDefaultRetryPolicies.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.retry;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import software.amazon.awssdk.RetryableException;
+import software.amazon.awssdk.http.HttpStatusCodes;
+
+public class SdkDefaultRetryPolicies {
+
+    /**
+     * When throttled retries are enabled, each retry attempt will consume this much capacity.
+     * Successful retry attempts will release this capacity back to the pool while failed retries
+     * will not.  Successful initial (non-retry) requests will always release 1 capacity unit to the
+     * pool.
+     */
+    public static final int THROTTLED_RETRY_COST = 5;
+
+    /**
+     * When throttled retries are enabled, this is the total number of subsequent failed retries
+     * that may be attempted before retry capacity is fully drained.
+     */
+    public static final int THROTTLED_RETRIES = 100;
+
+    public static final Integer BASE_DELAY = 100;
+
+    public static final Integer MAX_BACKOFF_IN_MILLIS = 20_000;
+
+    public static final Integer DEFAULT_NUM_RETRIES = 3;
+
+    public static final Set<String> THROTTLING_ERROR_CODES;
+    public static final Set<String> CLOCK_SKEW_ERROR_CODES;
+    public static final Set<String> RETRYABLE_ERROR_CODES;
+
+    public static final Set<Integer> RETRYABLE_STATUS_CODES;
+    public static final Set<Class> RETRYABLE_EXCEPTIONS;
+
+    static {
+        THROTTLING_ERROR_CODES = new HashSet<>();
+        THROTTLING_ERROR_CODES.add("Throttling");
+        THROTTLING_ERROR_CODES.add("ThrottlingException");
+        THROTTLING_ERROR_CODES.add("ProvisionedThroughputExceededException");
+        THROTTLING_ERROR_CODES.add("SlowDown");
+        THROTTLING_ERROR_CODES.add("TooManyRequestsException");
+        THROTTLING_ERROR_CODES.add("RequestLimitExceeded");
+        THROTTLING_ERROR_CODES.add("BandwidthLimitExceeded");
+        THROTTLING_ERROR_CODES.add("RequestThrottled");
+
+        CLOCK_SKEW_ERROR_CODES = new HashSet<>();
+        CLOCK_SKEW_ERROR_CODES.add("RequestTimeTooSkewed");
+        CLOCK_SKEW_ERROR_CODES.add("RequestExpired");
+        CLOCK_SKEW_ERROR_CODES.add("InvalidSignatureException");
+        CLOCK_SKEW_ERROR_CODES.add("SignatureDoesNotMatch");
+        CLOCK_SKEW_ERROR_CODES.add("AuthFailure");
+        CLOCK_SKEW_ERROR_CODES.add("RequestInTheFuture");
```

Comment:
These are kind of AWS specific. Do we have a plan for pulling them out of core?

Reply:
+1 I know this would expand the scope quite a bit but long term I think we want to track these via traits and generate service specific retry conditions.

Reply:
Generating retry policies per service is definitely something that we should do and I think this PR makes this easy.

I imagine always having some base set like this but we should generate custom policies per service based on some model and then wire those up in the default client builder. That's why I removed the DynamoDB one from here. They should be generated within the service module and wire up on generation.

We should take another look at the proposed JSON retry schema and try and get that in. We don't have to do that now though and I don't believe we will be blocked from doing so. It might mean the retry policy changes for a service but I don't think that is not backwards compatible as it should be additive from the default policy.